internal/virtualcircuit/create.go

```diff
@@ -0,0 +1,157 @@
+package virtualcircuit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	"github.com/spf13/cobra"
+)
+
+type vcParamBuilder interface {
+	SetName(string)
+	SetDescription(string)
+	SetTags([]string)
+}
+
+// compile-time assertion that parameter types implement vcParamBuilder
+var (
+	_ vcParamBuilder = (*metal.VlanVirtualCircuitCreateInput)(nil)
+	_ vcParamBuilder = (*metal.VrfVirtualCircuitCreateInput)(nil)
+	_ vcParamBuilder = (*metal.VlanVirtualCircuitUpdateInput)(nil)
+	_ vcParamBuilder = (*metal.VrfVirtualCircuitUpdateInput)(nil)
+)
+
+func setParams(p vcParamBuilder, name, description string, tags []string) {
+	if name > "" {
+		p.SetName(name)
+	}
+	if description > "" {
+		p.SetDescription(description)
+	}
+
+	if len(tags) > 0 {
+		p.SetTags(tags)
+	}
+}
+
+func (c *Client) Create() *cobra.Command {
+	var (
+		connectionID string
```

Comment:
This subcommand would take so many flags as input. Should we figure out another way. Probably take input a config file which has those data and we take the flags only for very necessary information

Reply:
A config file might be an interesting option, but we should not introduce that in this PR.

A different way to address the complexity of this subcommand would be to create separate subcommands for each virtual circuit type instead of having one that handles all types, but that gets away from how the API is structured so I think it's fine to have one subcommand for now.